File: frontend/src/app/shared/models/gateway/presence-realtime-event.ts
export interface PresenceRealtimeEvent {
  tipo: 'PRESENCE_ONLINE' | 'PRESENCE_OFFLINE' | 'PRESENCE_DISPONIBILIDAD_ACTUALIZADA' | 'PRESENCE_EXPIRED';
  empleadoId: number;
  nombreCompleto?: string | null;
  roles: string[];
  disponibilidad?: string | null;
  lastSeen?: string | null;
  online: boolean;
  source: 'ONLINE_ENDPOINT' | 'OFFLINE_ENDPOINT' | 'HEARTBEAT_RECOVERY' | 'DISPONIBILIDAD_ENDPOINT' | 'REDIS_TTL';
  occurredAt?: string | null;
}

File: gateway-service/src/main/java/pe/albrugroup/gateway_service/presence/PresenceRealtimeEvent.java
package pe.albrugroup.gateway_service.presence;

import lombok.Builder;

import java.time.Instant;
import java.util.List;

@Builder
public record PresenceRealtimeEvent(
        String tipo,
        Long empleadoId,
        String nombreCompleto,
        List<String> roles,
        String disponibilidad,
        Instant lastSeen,
        boolean online,
        String source,
        Instant occurredAt
) {
}
